### PR TITLE
add depositETH test for optimismPortal and optimismPortal2

### PR DESCRIPTION
**Pull request by AmadiMichael**
add depositETH test for optimismPortal and optimismPortal2

### CODE REVIEW

File: packages/contracts-bedrock/test/L1/OptimismPortal.t.sol [hunk 1]
```diff
@@ -570,6 +570,47 @@ contract OptimismPortal_Test is CommonTest {
...
+        optimismPortal.donateETH{ value: _amount }();
+        VmSafe.AccountAccess[] memory accountAccesses = vm.stopAndReturnStateDiff();
+
+        // not necessary since it's checked below
+        assertEq(address(optimismPortal).balance, preBalance + _amount);
+
+        // 0 for extcodesize of proxy before being called by this test,
+        // 1 for the call to the proxy by the pranked address
+        // 2 for the delegate call to the impl by the proxy
+        assertEq(accountAccesses.length, 3);
```
Comment (smartcontracts) [inline packages/contracts-bedrock/test/L1/OptimismPortal.t.sol:591]: We can probably do this without the state diff recording, right? 

Really just need to check that the balance increased

Comment (AmadiMichael) [reply packages/contracts-bedrock/test/L1/OptimismPortal.t.sol:591]: yeah, maybe, added it as an extra check to ensure nothing happens (read or write) regarding storage in the contract

Comment (AmadiMichael) [reply packages/contracts-bedrock/test/L1/OptimismPortal.t.sol:591]: should it be removed then?

Comment (smartcontracts) [reply packages/contracts-bedrock/test/L1/OptimismPortal.t.sol:591]: Honestly it's fine, this seems reasonable as-is.

**Review by smartcontracts — APPROVED** [review]

**Pull request merged**